### PR TITLE
Check for stocklines when creating stocktake

### LINKED ISSUE #889: Unable to create stocktake with items
..when there are a lot of items.

Using the recent demo file from Richard, which has a lot of items in it, when I create a stocktake, and tick items - the stocktake is created but is empty 🤷 

Also, when I try to add items, I can only see the first n items ( up to `D` in this datafile )

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #889 

The `stockLines` array exists on the items.. but is empty. In which case the stocktake line isn't added 🙄

### CODE REVIEW

File: client/packages/inventory/src/Stocktake/api/api.ts [hunk 1]
```diff
@@ -225,11 +225,9 @@ const getInsertStocktakeLines = (
   items: { itemId: string; stockLines?: StockLineFragment[] | undefined }[]
 ) => {
   const insertStocktakeLines = [] as InsertStocktakeLineInput[];
-
   items.forEach(item => {
     const { itemId, stockLines } = item;
```
Comment (clemens-msupply) [inline client/packages/inventory/src/Stocktake/api/api.ts:229]: So you either
- add lines from existing stocklines
- add line for an item that doesn't has an existing stockline

right?

can that better be express through:
``` typescript
type Item = {
  itemId?: string,
  stockLines?: StockLineFragment[],
}

// then 
if (item.stockLines) {...} else if (item.ItemId) {...}

// or even define it like
type Item = string | StockLineFragment[];
```

(had some trouble figure out the logic, maybe comment helps as well)

File: client/packages/system/src/Item/api/operations.graphql [hunk 2]
```diff
@@ -139,11 +139,14 @@ query itemsWithStats(
...
   $isDesc: Boolean
   $filter: ItemFilterInput
+  $first: Int
+  $offset: Int
 ) {
   items(
     storeId: $storeId
     sort: { key: $key, desc: $isDesc }
     filter: $filter
+    page: { first: $first, offset: $offset }
```
Comment (clemens-msupply) [inline client/packages/system/src/Item/api/operations.graphql:149]: that wasn't needed for this PR or did I miss it?

**Review by clemens-msupply — APPROVED** [review]
Thanks looks good

File: client/packages/inventory/src/Stocktake/api/api.ts [hunk 3]
```diff
@@ -225,11 +225,9 @@ const getInsertStocktakeLines = (
   items: { itemId: string; stockLines?: StockLineFragment[] | undefined }[]
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply client/packages/inventory/src/Stocktake/api/api.ts:225]: 🙏 it confused me, why I would have done that.. but I see that it's existing code! 😆 phew! ( I probably did do it.. but some time ago )

File: client/packages/system/src/Item/api/operations.graphql (hunk 2 above)
Comment (mark-prins) [reply client/packages/system/src/Item/api/operations.graphql:149]: sorry, offroad, sort of. The list of items shown was being truncated by the server default pagination limit, even though the client was at some point specifying a much higher limit. This meant on the datafile being used, only items up to `c` were shown

File: client/packages/inventory/src/Stocktake/api/api.ts (hunk 1 above)
Comment (mark-prins) [reply client/packages/inventory/src/Stocktake/api/api.ts:229]: yes, that's right - it creates stocktake lines for each stock line, if there are stocklines.
If no stocklines, you still want a placeholder stocktake line, so create one which references the item.

I've added the `Item` typing and added comments. Hope it is clearer!

**Pull request merged**